### ISSUE TITLE
iscriptevaluator.exe syntax changed

### ISSUE THREAD

**Issue by frostworx** [sonic2kk/steamtinkerlaunch#270]
As already mentioned [here](https://github.com/frostworx/steamtinkerlaunch/issues/269) the [Steam-First-Time-Setup](https://github.com/frostworx/steamtinkerlaunch/wiki/Steam-First-Time-Setup) segfaults with newer proton/wine versions.
No idea if this only happens when launching it "manually" or if this is really an upstream bug (or if it is caused by something else).

To be precise _(for search engines)_ the iscriptevaluator.exe segfaults with this output:

```
src\vstdlib\strtools_unicode.cpp (247) : Assertion Failed: invalid input pointer
src\vstdlib\strtools_unicode.cpp (247) : Assertion Failed: invalid input pointer
src\clientdll\installscript_win32.cpp (2231) : Assertion Failed: pchInstallPath
wine: Unhandled page fault on read access to 00000000 at address 00408470 (thread 0194), starting debugger...
Unhandled exception: page fault on read access to 0x00000000 in 32-bit code (0x00408470).
Register dump:
 CS:0023 SS:002b DS:002b ES:002b FS:0063 GS:006b
 EIP:00408470 ESP:006ef4f4 EBP:006ef620 EFLAGS:00010202(  R- --  I   - - - )
 EAX:00000001 EBX:00513210 ECX:00000000 EDX:00000001
 ESI:00000000 EDI:00513210
Stack dump:
0x006ef4f4:  0044e27e 00000000 0000002d 006ef51c
0x006ef504:  006ef520 006ef524 00443c56 00513210
0x006ef514:  006ef528 0043b24c 00443c56 0050d268
0x006ef524:  006ef560 0043b14f 00000003 0043b12e
0x006ef534:  3250f1de 00000000 00000003 00513210
0x006ef544:  00000000 006ef534 006ef570 006ef5d4
Backtrace:
=>0 0x00408470 EntryPoint+0xffffffff() in iscriptevaluator (0x006ef620)
  1 0x00406937 EntryPoint+0xffffffff() in iscriptevaluator (0x006efd80)
  2 0x00408e98 EntryPoint+0xffffffff() in iscriptevaluator (0x006efed4)
  3 0x00408d25 EntryPoint+0xffffffff() in iscriptevaluator (0x006eff30)
  4 0x7b62dc60 EntryPoint+0xffffffff() in kernel32 (0x006eff48)
  5 0x7bc57b67 EntryPoint+0xffffffff() in ntdll (0x006eff5c)
  6 0x7bc57db0 EntryPoint+0xffffffff() in ntdll (0x006effec)
0x00408470 EntryPoint+0xffffffff in iscriptevaluator: movb	0x0(%ecx),%al
Modules:
Module	Address			Debug info	Name (23 modules)
PE	  400000-  4d8000	Export          iscriptevaluator
PE	61740000-61782000	Deferred        advapi32
PE	62b80000-62b88000	Deferred        api-ms-win-core-localization-l1-2-1
PE	62fc0000-6304d000	Deferred        rpcrt4
PE	63480000-6348d000	Deferred        version
PE	63bc0000-63bd9000	Deferred        shcore
PE	64a40000-64a94000	Deferred        shlwapi
PE	64ec0000-64fed000	Deferred        oleaut32
PE	67840000-6784a000	Deferred        psapi
PE	68500000-68559000	Deferred        combase
PE	6a400000-6a530000	Deferred        ole32
PE	6ac80000-6ac87000	Deferred        api-ms-win-core-fibers-l1-1-1
PE	6bc00000-6bc2c000	Deferred        sechost
PE	6bcc0000-6bd3f000	Deferred        setupapi
PE	6c9c0000-6cb1c000	Deferred        gdi32
PE	6de80000-6de88000	Deferred        api-ms-win-core-synch-l1-2-0
PE	6ed00000-6ef18000	Deferred        user32
PE	70b40000-70c12000	Deferred        ucrtbase
PE	71200000-7121c000	Deferred        imm32
PE	7b000000-7b0e6000	Deferred        kernelbase
PE	7b600000-7b819000	Export          kernel32
PE	7bc00000-7bca0000	Export          ntdll
PE	7dfe0000-7e8b7000	Deferred        shell32
Threads:
process  tid      prio (all id:s are in hex)
00000030 services.exe
	00000034    0
	00000038    0
	0000004c    0
	00000050    0
	0000005c    0
	0000006c    0
	00000084    0
	00000090    0
	000000fc    0
	0000013c    0
	00000140    0
0000003c winedevice.exe
	00000040    0
	00000054    0
	00000058    0
	00000060    0
00000064 plugplay.exe
	00000068    0
	00000070    0
	00000074    0
	00000078    0
	00000098    0
	0000009c    0
0000007c winedevice.exe
	00000080    0
	00000088    0
	0000008c    0
	00000094    0
	000000a0    0
	000000a4    0
	000000a8    0
	000000ac    0
	000000b0    0
	000000b4    0
	000000b8    0
	000000bc    0
	000000c0    0
	000000c4    0
	000000c8    0
	000000cc    0
	000000d0    0
	000000d4    0
	000000d8    0
	000000dc    0
	000000e0    0
	000000e4    0
	000000e8    0
	000000ec    0
	000000f0    0
000000f4 svchost.exe
	000000f8    0
	00000100    0
	00000104    0
00000118 explorer.exe
	0000011c    0
	00000128    0
	0000012c    0
00000134 rpcss.exe
	00000138    0
	00000144    0
	00000148    0
	0000014c    0
	00000150    0
	00000154    0
	00000158    0
00000180 steam.exe
	00000184    0
00000188 conhost.exe
	0000018c    0
00000190 (D) Z:\home\frostwork\.steam\root\legacycompat\iscriptevaluator.exe
	00000194    0 <==
	000001a0    0
System information:
    Wine build: wine-6.3
    Platform: i386 (WOW64)
    Version: Windows 10
    Host system: Linux
    Host version: 5.13.4-xanmod1-1
```

**Comment by frostworx** [comment]
This and [write access blocks to files](https://github.com/frostworx/steamtinkerlaunch/issues/271) could become problematic if they stay permanently.

**Comment by frostworx** [comment]
The string `foreign_install_path` is no longer used in the temporary `evaluatorscript_XXX.vdf` files.
Instead we have `linux_install_path` now, which even directly points to the absolute `Steamworks Shared` directory. nice!

**Comment by frostworx** [comment]
Fixed in release [v6.13](https://github.com/frostworx/steamtinkerlaunch/releases/tag/v6.13)
